probe-rs/src/debug/variable.rs

```diff
@@ -655,214 +699,239 @@ impl Variable {
-                        compound_value = format!("{compound_value}{line_start}{post_fix}");
-                    }
-                    compound_value
-                }
+        // Make sure we can safely unwrap() children.
+        Some(match self.type_name.inner() {
+            VariableType::Pointer(_) => {
+                format_pointer_value(variable_cache, indentation, first_child)
             }
-        }
+            VariableType::Array { .. } => {
+                format_array_value(variable_cache, indentation, children, &type_name)
+            }
+            VariableType::Struct(name) if name == "Some" || name == "Ok" || name == "Err" => {
+                format_struct_value(variable_cache, indentation, children, &type_name)
+            }
+            _ if first_child.is_none() => {
+                // This is a struct with no children, so just print the type name.
+                // This is for example the None value of an Option or the empty type ().
+                type_name
+            }
+            _ if matches!(
+                self.name,
+                VariableName::StaticScopeRoot
+                    | VariableName::LocalScopeRoot
+                    | VariableName::RegistersRoot
+            ) =>
+            {
+                format_root_value(variable_cache, indentation, children, &type_name)
+            }
+            _ => format_default_value(
+                variable_cache,
+                indentation,
+                &self.name,
+                children,
+                &type_name,
+                show_name,
+            ),
+        })
     }
 
     /// Calculate the memory range that contains the value of this variable.
+    ///
     /// If the location and/or byte size is not known, then return None.
-    /// Note: We don't do any validation of the memory range here, and leave it up to the caller to
-    /// validate the memory ranges before attempting to read them.
+    /// Note: We don't do any validation of the memory range here and leave it
+    /// up to the caller to validate the memory ranges before attempting to read
+    /// them.
     pub fn memory_range(&self) -> Option<Range<u64>> {
-        if let VariableLocation::Address(address) = self.memory_location {
-            if self.byte_size.is_some_and(|byte_size| byte_size.is_zero()) {
-                // This can happen for instance with empty arrays, and even though we don't need to read the
-                // array data, we need to read the array structure type.
-                Some(address..address + 4)
-            } else {
-                self.byte_size
-                    .map(|byte_size| address..(address + byte_size))
-            }
+        let VariableLocation::Address(address) = self.memory_location else {
+            return None;
+        };
+
+        self.byte_size.map(|byte_size| {
+            byte_size
+                .is_zero()
+                .then(|| address..address + 4)
```

Comment:
I'm not sure why Clippy doesn't scream about trivial closures here, these are better as `then_some()` and `unwrap_or`. It's not like creating a range is computationally expensive.

That said, this is pretty complex when what you want is 
```rust
self.byte_size.map(|mut byte_size| {
    if byte_size == 0 { byte_size = 4; }
    address..address + byte_size
}
```